[vNext Tokens] Convert SegmentedControl from UIControl to UIView

huwilkes opened the pull request:
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

The change from UIControl to UIView will be breaking, as clients will now need to set the onSelectAction rather than adding a target.

The current SegmentedControl is a UIControl, and that means that keyboard navigation will only select the entire control. By changing it to a UIView, we can now select each individual button. I recorded the keyboard navigation videos with increased contrast to show where the focus was. We don't have focus colors from design right now, so I'll come back and update the colors to change once we have those. With this change, though, the control is at least usable with keyboard navigation.

### Verification

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| ![SegmentedControl_UIView_before](https://user-images.githubusercontent.com/67026548/173468085-eae71531-1fce-435b-9e05-0623110f7e75.gif) | ![SegmentedControl_UIView_after](https://user-images.githubusercontent.com/67026548/173468092-52c2478c-1c8c-437e-8bba-04e4f473ba25.gif) |
| ![SegmentedControl_Color_before](https://user-images.githubusercontent.com/67026548/173468109-4cbcb915-824a-4065-92c6-a87ac5a68895.gif) | ![SegmentedControl_Color_after](https://user-images.githubusercontent.com/67026548/173468119-cabec842-1482-45af-be50-7937bb01eaba.gif) |
| Crash? | ![SegmentedControl_Table_after](https://user-images.githubusercontent.com/67026548/173468128-2c71774e-91c7-482f-b4c9-f35c9d39e383.gif) |
| ![SegmentedControl_Date_before](https://user-images.githubusercontent.com/67026548/173468134-2db02b12-481d-44e4-bd35-17e3e5205faa.gif) | ![SegmentedControl_Date_after](https://user-images.githubusercontent.com/67026548/173468155-470134fe-00cc-424d-8c98-f1cc3fee1b70.gif) |
| ![SegmentedControl_Time_before](https://user-images.githubusercontent.com/67026548/173468161-b3e3ff9a-b39c-4ea6-bad7-09d1e898e7e7.gif) | ![SegmentedControl_Time_after](https://user-images.githubusercontent.com/67026548/173468196-8cb3ddfa-41e6-4c89-87e2-0948fd5c4f73.gif) |

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [x] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1006)

harrieshin commented on ios/FluentUI/SegmentedControl/SegmentedControl.swift:
```diff
@@ -47,6 +47,8 @@ open class SegmentedControl: UIControl, TokenizedControlInternal {
         }
     }
+    /// Action called when a segment is selected.
+    @objc public var onSelectAction: ((SegmentedControl, SegmentItem, Int) -> Void)?
```
better documentation what int means? but i don't quite understand why we need segmentedcoontrol as a param?

huwilkes replied:
The SegmentedControl is in there so it could be used in the same ways as sender used to. We could probably remove it, but our demo control uses the SegmentedControls to access the labels below them.

harrieshin commented on ios/FluentUI/SegmentedControl/SegmentedControl.swift:
```diff
@@ -7,7 +7,7 @@ import UIKit
 /// A styled segmented control that should be used instead of UISegmentedControl. It is designed to flex the button width proportionally to the control's width.
 @objc(MSFSegmentedControl)
-open class SegmentedControl: UIControl, TokenizedControlInternal {
+open class SegmentedControl: UIView, TokenizedControlInternal {
```
any voiceover code changes we need now that we are uiview rather than uicontrol?

huwilkes replied:
VoiceOver is still working fine, says "First, button, 1 of 3"

harrieshin commented:
> Please document some details about the public API change.

also make note about in the PR how this is breaking change so when we merge and bring release notes easier to capture all the breaking changes together.

huwilkes commented:
> > Please document some details about the public API change.
> 
> also make note about in the PR how this is breaking change so when we merge and bring release notes easier to capture all the breaking changes together.

Updated the description and added a API break label
